client/state/selectors/has-user-devices.js

```diff
@@ -6,9 +6,9 @@ import { isEmpty } from 'lodash';
 /**
  * Returns a boolean signifying whether there are user devices or not
  *
- * @param  {Object} state Global state tree
- * @return {Boolean}      true if the user has devices
+ * @param  {Object} userDevices user's devices slice of the state tree
+ * @return {Boolean}            true if the user has devices
  */
-export default function hasUserSettings( state ) {
-	return ! isEmpty( state.userDevices.items );
-}
+export const hasUserSettings = ( { userDevices } ) => ! isEmpty( userDevices );
```

Comment:
That selector is for the benefit of a single component and most likely will be removed in favor of `getUserDevices`. This is the code: https://github.com/Automattic/wp-calypso/pull/14304/files#diff-420918acb15a2d764268e62f6c43971b. It hasn't been refactored with the changes on this PR yet.

For the use case, there's no distinction between _we don't know_ and _we don't have any_ because for both cases the _devices_ column has to be hid. I'd like to keep it as it is and remove it on #14304 once we have this merged and I have a chance to rebase and refactor it.